src/stream/src/executor/aggregation/minput.rs

```diff
@@ -102,9 +104,26 @@ impl MaterializedInputState {
                     .unzip()
             };
 
-        let pk_len = pk_indices.len();
-        order_col_indices.extend(pk_indices.iter());
-        order_types.extend(itertools::repeat_n(OrderType::ascending(), pk_len));
+        if agg_call.distinct {
+            if version < PbAggNodeVersion::Issue12140 {
+                panic!(
+                    "RisingWave versions before issue #12140 is resolved has critical bug, you may need to re-create current MV to ensure correctness."
```

Comment:
```suggestion
                    "RisingWave versions before issue #12140 is resolved has critical bug, you must re-create current MV to ensure correctness."
```

Reply:
Will this called before initialization of the `SimplAggExecutor`? If so, then currently we have no way to drop it. 😕 Though this definitely needs improvement -> #12999.